### PR TITLE
chore(ci): update nightly

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Update the nightly version

Motivation and Context
---
one of the used dependencies uses a feature that only went stable in April, so we need to update the used version

How Has This Been Tested?
---
ran all tests

### CODE REVIEW

File: applications/tari_miner/src/stratum/controller.rs [hunk 1]
```diff
@@ -243,6 +243,7 @@ impl Controller {
             // problem with template
             let _result = self.send_message_get_job_template();
         } else {
+            // unhandled
```
Comment (brianp) [inline applications/tari_miner/src/stratum/controller.rs:246]: It only kinda bugs me that the comments aren't consistent 🙃 

What's the need for them all of a sudden though, all they failing a linter now or something?

File: base_layer/common_types/src/transaction.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@ pub enum TransactionStatus {
     /// This transaction was generated as part of importing a spendable unblinded UTXO
     Imported,
     /// This transaction is still being negotiated by the parties
+    #[default]
```
Comment (brianp) [inline base_layer/common_types/src/transaction.rs:27]: Neat! TIL.

File: comms/core/src/transports/memory.rs [hunk 3]
```diff
@@ -137,7 +137,7 @@ mod test {
 
     #[tokio::test]
     async fn simple_listen_and_dial() -> Result<(), ::std::io::Error> {
-        let t = MemoryTransport::default();
+        let t = MemoryTransport;
```
Comment (brianp) [inline comms/core/src/transports/memory.rs:140]: is this a new style / default? Feels foreign and anti-intuitive.

**Review by brianp — APPROVED** [review]
LGTM

Just to clarify the purpose of the PR is to bump the rust version, and that required syntax fixing for linters?

**Comment by SWvheerden** [comment]
Updates to clippy is what caused most of the format changes

File: comms/core/src/transports/memory.rs (hunk 3 above)
Comment (SWvheerden) [reply comms/core/src/transports/memory.rs:140]: Believe it is,

File: base_layer/common_types/src/transaction.rs (hunk 2 above)
Comment (SWvheerden) [reply base_layer/common_types/src/transaction.rs:27]: yeah same

File: applications/tari_miner/src/stratum/controller.rs (hunk 1 above)
Comment (SWvheerden) [reply applications/tari_miner/src/stratum/controller.rs:246]: yip cant have an empty else { }
tried to make them as far as I could see comment why its empty

Comment (AaronFeickert) [reply applications/tari_miner/src/stratum/controller.rs:246]: Aren't there macros intended to handle these cases?

Comment (SWvheerden) [reply applications/tari_miner/src/stratum/controller.rs:246]: You can, but having an actual comment in there is more useful than a macro bypassing this check.

**Pull request merged**